From 433fad370ddd30c43df281b40ff468a842bc1c70 Mon Sep 17 00:00:00 2001
From: Adam Conrad <adconrad@ubuntu.com>
Date: Thu, 6 Feb 2014 09:13:32 +0100
Subject: [PATCH] Fix int/bool confusion on big-endian architectures.

---
 qa/libcmis-c/test-document.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qa/libcmis-c/test-document.cxx b/qa/libcmis-c/test-document.cxx
index ccf2224..eb2c01c 100644
--- a/qa/libcmis-c/test-document.cxx
+++ b/qa/libcmis-c/test-document.cxx
@@ -43,7 +43,7 @@
 
 using namespace std;
 
-extern int isOutOfMemory;
+extern bool isOutOfMemory;
 
 namespace
 {
-- 
1.8.5.3

